File: src/main/java/xyz/emirdev/echogen/commands/MainCommand.java
package xyz.emirdev.echogen.commands;

import org.bukkit.command.CommandSender;

import com.mojang.brigadier.context.CommandContext;
import com.mojang.brigadier.tree.LiteralCommandNode;

import io.papermc.paper.command.brigadier.CommandSourceStack;
import io.papermc.paper.command.brigadier.Commands;
import net.kyori.adventure.text.minimessage.tag.resolver.Placeholder;
import xyz.emirdev.echogen.Echogen;
import xyz.emirdev.echogen.PluginCommand;

public class MainCommand extends PluginCommand {

    @Override
    public LiteralCommandNode<CommandSourceStack> getCommand() {
        return Commands.literal("echogen")
                .requires(ctx -> ctx.getSender().hasPermission("echogen.admin"))
                .then(Commands.literal("reload")
                        .executes(this::reload))
                .build();
    }

    private int reload(CommandContext<CommandSourceStack> ctx) {
        CommandSender sender = ctx.getSource().getSender();
        long startedAt = System.currentTimeMillis();

        sender.sendRichMessage("<aqua>Reloading Echogen configuration...");

        Echogen.get().reloadConfig();

        sender.sendRichMessage(
                "<aqua>Reloaded Echogen configuration in <u><ms></u>ms.",
                Placeholder.unparsed("ms", String.valueOf(System.currentTimeMillis() - startedAt)));

        return 1;
    }
}

File: src/main/kotlin/cat/emir/echogen/commands/MainCommand.kt
package cat.emir.echogen.commands

import cat.emir.echogen.Echogen
import com.mojang.brigadier.context.CommandContext

import io.papermc.paper.command.brigadier.CommandSourceStack
import cat.emir.echolib.command.PluginCommand
import cat.emir.echolib.sendLangMessage
import com.mojang.brigadier.builder.LiteralArgumentBuilder

class MainCommand(plugin: Echogen) : PluginCommand<Echogen>(plugin) {

    override fun getCommand(): LiteralArgumentBuilder<CommandSourceStack> {
        return command("echogen") {
            requires { it.sender.hasPermission("echogen.admin") }
            subcommand("reload") {
                executes(::reload)
            }
        }
    }

    fun reload(ctx: CommandContext<CommandSourceStack>): Int {
        val sender = ctx.source.sender

        sender.sendLangMessage("reload.start")
        val time = plugin.reload()
        sender.sendLangMessage("reload.end", listOf("ms" to time.toString()))

        return 1
    }
}
